cloudinit/distros/bsd.py

```diff
@@ -86,12 +88,14 @@ def package_command(self, command, args=None, pkgs=None):
         if pkgs is None:
             pkgs = []
 
-        # TODO neither freebsd nor netbsd handles a command 'upgrade'
```

Comment:
This comment still appears to be true; this PR is adding handling for `update`, not `upgrade`. And, as far as I can tell, `package_command` is not called with `"update"`:

```
cloudinit/config/cc_package_update_upgrade_install.py:92:            cloud.distro.package_command("upgrade")
cloudinit/config/cc_package_update_upgrade_install.py:93-        except Exception as e:
cloudinit/distros/debian.py:111:        self.package_command('install', pkgs=pkglist)
cloudinit/distros/debian.py:112-
cloudinit/distros/rhel.py:62:        self.package_command('install', pkgs=pkglist)
cloudinit/distros/rhel.py:63-
cloudinit/distros/gentoo.py:53:        self.package_command('', pkgs=pkglist)
cloudinit/distros/gentoo.py:54-
cloudinit/distros/arch.py:58:        self.package_command('', pkgs=pkglist)
cloudinit/distros/arch.py:59-
cloudinit/distros/bsd.py:79:        self.package_command('install', pkgs=pkglist)
cloudinit/distros/bsd.py:80-
cloudinit/distros/opensuse.py:72:        self.package_command(
cloudinit/distros/opensuse.py:73-            'install',
```

Or, for that matter, with `"remove"`. Am I missing something here?

Reply:
I put back the comment, but on top of the file, for consistency. The update call is done here: https://github.com/canonical/cloud-init/blob/master/cloudinit/distros/bsd.py#L78

Reply:
Aha, I'd missed that little bit of indirection, thanks for the pointer!

I think my point still stands: this method doesn't handle "upgrade" commands, so the TODO here should remain in place.

(For avoidance of doubt, in Ubuntu/Debian and therefore in cloud-init terms, "update" refers to refreshing the packaging system's metadata and "upgrade" refers to the operation of installing newer versions of already installed packages.)

Reply:
The TODO is still in the file, but at the top of the `bsd.py`. `pkg upgrade` is actually a thing on FreeBSD and I plan to implement it.

Reply:
Oh, my apologies!